From 18831596085a326f9d31a2ce0c82835a1381d8ed Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Tue, 26 Sep 2017 10:08:06 -0500
Subject: [PATCH 2683/2748] HADOOP-14891. Remove references to Guava
 Objects.toStringHelper. Contributed by Jonathan
 Eagles

(cherry picked from commit 0aac5af6e1a2b265cc8faa99b386276ed14faeda)
(cherry picked from commit 86e092029dd5b254efa220a3a85c41f56d797690)

Change-Id: Ice0591aecf00671cff62e125e43975bb2180878b
---
 .../org/apache/hadoop/metrics2/AbstractMetric.java |   11 +++++++----
 .../org/apache/hadoop/metrics2/MetricsTag.java     |   11 +++++++----
 .../metrics2/impl/AbstractMetricsRecord.java       |   20 +++++++++++++-------
 .../org/apache/hadoop/metrics2/impl/MsInfo.java    |   12 +++++++-----
 .../hadoop/metrics2/lib/MetricsInfoImpl.java       |   10 +++++++---
 .../hadoop/metrics2/lib/MetricsRegistry.java       |   13 +++++++++----
 .../hadoop/metrics2/source/JvmMetricsInfo.java     |   12 +++++++-----
 .../apache/hadoop/metrics2/util/MetricsCache.java  |   11 +++++++----
 8 files changed, 64 insertions(+), 36 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/AbstractMetric.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/AbstractMetric.java
index 6a11b87..510cee0 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/AbstractMetric.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/AbstractMetric.java
@@ -84,9 +84,12 @@ protected MetricsInfo info() {
   }
 
   @Override public String toString() {
-    return Objects.toStringHelper(this)
-        .add("info", info)
-        .add("value", value())
-        .toString();
+    StringBuilder sb = new StringBuilder(32);
+    sb.append(this.getClass().getSimpleName());
+    sb.append("{info=");
+    sb.append(info);
+    sb.append(", value=");
+    sb.append(value());
+    return sb.append('}').toString();
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsTag.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsTag.java
index e9e50a4..23ef086 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsTag.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsTag.java
@@ -80,9 +80,12 @@ public String value() {
   }
 
   @Override public String toString() {
-    return Objects.toStringHelper(this)
-        .add("info", info)
-        .add("value", value())
-        .toString();
+    StringBuilder sb = new StringBuilder(32);
+    sb.append(this.getClass().getSimpleName());
+    sb.append("{info=");
+    sb.append(info);
+    sb.append(", value=");
+    sb.append(value());
+    return sb.append('}').toString();
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/AbstractMetricsRecord.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/AbstractMetricsRecord.java
index 3684c7e..cdcde14 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/AbstractMetricsRecord.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/AbstractMetricsRecord.java
@@ -43,12 +43,18 @@
   }
 
   @Override public String toString() {
-    return Objects.toStringHelper(this)
-        .add("timestamp", timestamp())
-        .add("name", name())
-        .add("description", description())
-        .add("tags", tags())
-        .add("metrics", Iterables.toString(metrics()))
-        .toString();
+    StringBuilder sb = new StringBuilder(32);
+    sb.append(this.getClass().getSimpleName());
+    sb.append("{timestamp=");
+    sb.append(timestamp());
+    sb.append(", name=");
+    sb.append(name());
+    sb.append(", description=");
+    sb.append(description());
+    sb.append(", tags=");
+    sb.append(tags());
+    sb.append(", metrics=");
+    sb.append(Iterables.toString(metrics()));
+    return sb.append('}').toString();
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/MsInfo.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/MsInfo.java
index 782f755..c66160f 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/MsInfo.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/MsInfo.java
@@ -18,8 +18,6 @@
 
 package org.apache.hadoop.metrics2.impl;
 
-import com.google.common.base.Objects;
-
 import org.apache.hadoop.classification.InterfaceAudience;
 import org.apache.hadoop.metrics2.MetricsInfo;
 
@@ -48,8 +46,12 @@
   }
 
   @Override public String toString() {
-    return Objects.toStringHelper(this)
-        .add("name", name()).add("description", desc)
-        .toString();
+    StringBuilder sb = new StringBuilder(32);
+    sb.append(this.getClass().getSimpleName());
+    sb.append("{name=");
+    sb.append(name());
+    sb.append(", description=");
+    sb.append(desc);
+    return sb.append('}').toString();
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsInfoImpl.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsInfoImpl.java
index dfb6c35..79bba1f 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsInfoImpl.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsInfoImpl.java
@@ -55,8 +55,12 @@
   }
 
   @Override public String toString() {
-    return Objects.toStringHelper(this)
-        .add("name", name).add("description", description)
-        .toString();
+    StringBuilder sb = new StringBuilder(32);
+    sb.append(this.getClass().getSimpleName());
+    sb.append("{name=");
+    sb.append(name);
+    sb.append(", description=");
+    sb.append(description);
+    return sb.append('}').toString();
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsRegistry.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsRegistry.java
index 4b561f2..88c8789 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsRegistry.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsRegistry.java
@@ -22,7 +22,6 @@
 import java.util.Map;
 
 import com.google.common.collect.Maps;
-import com.google.common.base.Objects;
 
 import org.apache.hadoop.classification.InterfaceAudience;
 import org.apache.hadoop.classification.InterfaceStability;
@@ -403,8 +402,14 @@ public synchronized void snapshot(MetricsRecordBuilder builder, boolean all) {
   }
 
   @Override public String toString() {
-    return Objects.toStringHelper(this)
-        .add("info", metricsInfo).add("tags", tags()).add("metrics", metrics())
-        .toString();
+    StringBuilder sb = new StringBuilder(32);
+    sb.append(this.getClass().getSimpleName());
+    sb.append("{info=");
+    sb.append(metricsInfo);
+    sb.append(", tags=");
+    sb.append(tags());
+    sb.append(", metrics=");
+    sb.append(metrics());
+    return sb.append('}').toString();
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetricsInfo.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetricsInfo.java
index 010ff64..e99f24d 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetricsInfo.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetricsInfo.java
@@ -18,8 +18,6 @@
 
 package org.apache.hadoop.metrics2.source;
 
-import com.google.common.base.Objects;
-
 import org.apache.hadoop.classification.InterfaceAudience;
 import org.apache.hadoop.metrics2.MetricsInfo;
 
@@ -61,8 +59,12 @@
   @Override public String description() { return desc; }
 
   @Override public String toString() {
-  return Objects.toStringHelper(this)
-      .add("name", name()).add("description", desc)
-      .toString();
+    StringBuilder sb = new StringBuilder(32);
+    sb.append(this.getClass().getSimpleName());
+    sb.append("{name=");
+    sb.append(name());
+    sb.append(", description=");
+    sb.append(desc);
+    return sb.append('}').toString();
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/MetricsCache.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/MetricsCache.java
index efcb286..9c7935b 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/MetricsCache.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/MetricsCache.java
@@ -31,7 +31,6 @@
 import org.apache.hadoop.metrics2.MetricsRecord;
 import org.apache.hadoop.metrics2.MetricsTag;
 
-import com.google.common.base.Objects;
 import com.google.common.collect.Maps;
 
 /**
@@ -127,9 +126,13 @@ public AbstractMetric getMetricInstance(String key) {
     }
 
     @Override public String toString() {
-      return Objects.toStringHelper(this)
-          .add("tags", tags).add("metrics", metrics)
-          .toString();
+      StringBuilder sb = new StringBuilder(32);
+      sb.append(this.getClass().getSimpleName());
+      sb.append("{tags=");
+      sb.append(tags);
+      sb.append(", metrics=");
+      sb.append(metrics);
+      return sb.append('}').toString();
     }
   }
 
-- 
1.7.9.5

